src/tests/integration/thrift/user.thrift

```diff
@@ -1,8 +1,47 @@
 namespace java com.test.user
 namespace js com.test.user
 
+include "shared.thrift"
+include "common/common.thrift"
+
+typedef common.CommonStruct CommonStruct
```

Comment:
This is what requires the hack. Linking `CommonStruct` -> `common.CommonStruct` -> `__ROOT_NAMESPACE__.SharedStruct` works on it's own. When resolving the `Code` substruct of the `SharedStruct` it wouldn't have the `__ROOT_NAMESPACE__` prefix because of how the `resolveIdentifierDefinition()` function is working. 

If we drop this support case the code is a lot simpler.